addons/escoria-core/game/core-scripts/esc/types/esc_dialog.gd

```diff
@@ -95,4 +95,6 @@ func run():
 		rc = yield(rc, "completed")
 	if rc != ESCExecution.RC_CANCEL:
 		return self.run()
+	# make sure we are resetting GAME_STATE to prevent the game from hanging
+	escoria.current_state = escoria.GAME_STATE.DEFAULT
```

Comment:
It might make some sense to store the previous state as a local variable prior to setting the state to `DIALOG` and then use that saved state to restore to here.

Reply:
I thought about that. This does the same thing the `say` command does. I preferred to not over complicate things.

Reply:
for reference: here
https://github.com/godot-escoria/escoria-demo-game/blob/998c9e2535f27889b8a9dbe85af27b726ff7d319/addons/escoria-core/game/core-scripts/esc/commands/say.gd#L118

Reply:
State needs to be encapsulated. Changing it in one place and then expecting/hoping another, external call will otherwise change it isn't a good idea.

And if after restoring the state something else breaks as a result, we should deal with it then. Leaking state changes is a code smell and I can't find anything in the codebase saying that it has to be that way.